src/one/nio/serial/Serializer.java

```diff
@@ -133,6 +137,15 @@ public static int sizeOf(Object obj) throws IOException {
     }
 
     public static Object deserialize(byte[] data) throws IOException, ClassNotFoundException {
-        return new DeserializeStream(data).readObject();
+        try {
+            return new DeserializeStream(data).readObject();
+        } catch (SerializerNotFoundException e) {
```

Comment:
The difference is that RpcClient is an outer layer. It does not break into streams, and handling deserialization errors is a part of the protocol, But inside serialization engine manipulating its own errors looks like a hack.